R/zchunk_LB122.LC_R_Cropland_Yh_GLU.R

```diff
@@ -43,91 +47,603 @@ module_aglu_LB122.LC_R_Cropland_Yh_GLU_DISABLED <- function(command, ...) {
+      # store in a table of OtherArableLand by region-glu-year, including historical years:
+      L122.LC_bm2_R_OtherArableLand_Yh_GLU
+
+
+    # Lines 202-205 in original file
+    # printlog( "All other cropland uses are zero in the pre-aglu years" )
+    # Specifically, ExtraCropLand is expanded to include historical years in each region-glu, with a value of 0.
+    # Take ExtraCropLand by region-glu-year:
+    L122.LC_bm2_R_ExtraCropLand_Y_GLU %>%
+      # expand to include history years and fill in those values to be 0:
+      tidyr::complete(year = c(PREAGLU_YEARS, AGLU_HISTORICAL_YEARS),
+                      nesting(GCAM_region_ID, GLU, Land_Type),
+                      fill = list(value = 0)) ->
+      # store in a table of ExtraCropland by region-glu-year, including historical years:
+      L122.LC_bm2_R_ExtraCropLand_Yh_GLU
+
+
+    # Lines 207-210 in original file
+    # old comment: Harvested cropland history
+    # HarvCropLand is expanded to include historical years for each region-glu-commodity, with a vlaue of 0.
+    # Take HarvCropLand by region-commodity-glu-year:
+    L122.LC_bm2_R_HarvCropLand_C_Y_GLU %>%
+      # expand to include history years and fill in those values to be 0:
+      tidyr::complete(year = c(PREAGLU_YEARS, AGLU_HISTORICAL_YEARS),
+                      nesting(GCAM_region_ID, GCAM_commodity, GLU, Land_Type),
+                      fill = list(value = 0)) ->
+      # store in a table of HarvCropland by region-commodity-glu-year, including historical years:
+      L122.LC_bm2_R_HarvCropLand_C_Yh_GLU
+
+    # Lines 212-217 in original file
+    # old comment: Combine crop types to get land use history by all harvested cropland.
+    # Use the table of HarvCropland by region-commodity-glu-year, including historical years, L122.LC_bm2_R_HarvCropLand_C_Yh_GLU,
+    # and aggregate across commodity to arrive at a table of All Harvested Cropland by region-glu-year, including historical years.
+    # Take HarvCropland by region-commodity-glu-year:
+    L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
+      # remove commodity because that is what aggregate over:
+      select(-GCAM_commodity) %>%
+      # group by region-glu-year to aggregate:
+      group_by(GCAM_region_ID, GLU, Land_Type, year) %>%
+      # aggregate:
+      summarise(value = sum(value)) ->
+      # store in a table of all HarvCropland by region-glu-year, including historical years:
+      L122.LC_bm2_R_HarvCropLand_Yh_GLU
+
+    # remove the now-unneeded Land_Type identifier from HarvCropland by region-commodity-glu-year:
+    L122.LC_bm2_R_HarvCropLand_C_Yh_GLU %>%
+      select(-Land_Type) ->
+      L122.LC_bm2_R_HarvCropLand_C_Yh_GLU
+
+
+    # Force-Convert years and GCAM_region_IDs in some problematic outputs to integers:
+    L122.LC_bm2_R_ExtraCropLand_Yh_GLU %>%
+      ungroup() %>%
+      mutate(year = as.integer(year),
+             GCAM_region_ID = as.integer(GCAM_region_ID),
+             value = format(value, digits = 12)) ->
```

Comment:
This statement causes `value` to be converted into a character, which is going to cause problems downstream.  Any subsequent processing that tries to use this data is going to have to convert it back to numeric.

The right thing to do here is add `FLAG_PROTECT_FLOAT` to this table, which will cause this character conversion to be done just before write-out.  If the current number of digits in that character conversion isn't enough; we can always increase it.